src/core/meshcop/tcat_agent.hpp

```diff
@@ -0,0 +1,398 @@
+        kTlvGetProvissioningTlvs                   = 36, ///< TCAT provisioning TLVs query TLV
+        kTlvGetCommissionerCertificate             = 37, ///< TCAT commissioner certificate query TLV
+        kTlvGetDiagnosticTlvs                      = 38, ///< TCAT diagnostics TLVs query TLV
+        kTlvStartThreadInterface                   = 39, ///< TCAT start thread interface request TLV
+        kTlvStopThreadInterface                    = 40, ///< TCAT stop thread interface request TLV
+
+        // Command Class Extraction
+        kTlvGetActiveOperationalDataset            = 64, ///< TCAT active oerational dataset query TLV
+        kTlvGetActiveOperationalDatasetAlternative = 65, ///< TCAT active oerational dataset alternative #1 query TLV
+
+        // Command Class Decommissioning
+        kTlvDecommission = 96, ///< TCAT decommission request TLV
+
+        // Command Class Application
+        kTlvSelectApplicationLayerUdp = 128, ///< TCAT select UDP protocol application layer request TLV
+        kTlvSelectApplicationLayerTcp = 129, ///< TCAT select TCP protocol application layer request TLV
+        kTlvSendApplicationData       = 130, ///< TCAT send application data TLV
+        kTlvSendVendorSpecificData    = 159, ///< TCAT send vendor specific command or data TLV
+
+        // Command Class CCM
+        kTlvSetLDevIdOperationalCert = 160, ///< TCAT LDevID operational certificate TLV
+        kTlvSetLDevIdPrivateKey      = 161, ///< TCAT LDevID operational certificate pricate key TLV
+        kTlvSetDomainCaCert          = 162, ///< TCAT domain CA certificate TLV
+    };
+
+    /**
+     * TCAT Response Types.
+     *
+     */
+    enum StatusCode : uint8_t
+    {
+        kStatusSuccess      = OT_TCAT_STATUS_SUCCESS,       ///< Command or request was successfully processed
+        kStatusUnsupported  = OT_TCAT_STATUS_UNSUPPORTED,   ///< Requested command or received TLV is not supported
+        kStatusParseError   = OT_TCAT_STATUS_PARSE_ERROR,   ///< Request / command could not be parsed correctly
+        kStatusValueError   = OT_TCAT_STATUS_VALUE_ERROR,   ///< The value of the transmitted TLV has an error
+        kStatusGeneralError = OT_TCAT_STATUS_GENERAL_ERROR, ///< An error not matching any other category occurred
+        kStatusBusy         = OT_TCAT_STATUS_BUSY,          ///< Command cannot be executed because the resource is busy
+        kStatusUndefined    = OT_TCAT_STATUS_UNDEFINED,     ///< The requested value, data or service is not defined
+                                                            ///< (currently) or not present
+        kStatusHashError = OT_TCAT_STATUS_HASH_ERROR, ///< The hash value presented by the commissioner was incorrect
+        kStatusUnauthorized =
+            OT_TCAT_STATUS_UNAUTHORIZED, ///< Sender does not have sufficient authorization for the given command
+    };
+
+    /**
+     * Represents TCAT message type.
+     *
+     */
+    enum TcatMessageType : uint8_t
```

Comment:
see my earlier review comments on the #define items that are in this enum.

Reply:
fixed.